lib/Map/ImageryProvider/ProtomapsImageryProvider.ts

```diff
@@ -498,47 +514,48 @@ export default class ProtomapsImageryProvider
         (r) => r.dataLayer
       );
 
-      return filterOutUndefined(
-        this.view
-          .queryFeatures(
-            CesiumMath.toDegrees(longitude),
-            CesiumMath.toDegrees(latitude),
-            level
+      this.view
+        .queryFeatures(
+          CesiumMath.toDegrees(longitude),
+          CesiumMath.toDegrees(latitude),
+          level
+        )
+        .forEach((f) => {
+          // Only create FeatureInfo for visible features with properties
+          if (
+            !f.feature.props ||
+            isEmpty(f.feature.props) ||
+            !renderedLayers.includes(f.layerName)
           )
-          .map((f) => {
-            // Only create FeatureInfo for visible features with properties
-            if (
-              !f.feature.props ||
-              isEmpty(f.feature.props) ||
-              !renderedLayers.includes(f.layerName)
-            )
-              return;
-
-            const featureInfo = new ImageryLayerFeatureInfo();
-
-            // Add Layer name property
-            featureInfo.properties = Object.assign(
-              { [LAYER_NAME_PROP]: f.layerName },
-              f.feature.props ?? {}
-            );
-            featureInfo.position = new Cartographic(longitude, latitude);
+            return;
 
-            featureInfo.configureDescriptionFromProperties(f.feature.props);
-            featureInfo.configureNameFromProperties(f.feature.props);
+          const featureInfo = new ImageryLayerFeatureInfo();
+
+          // Add Layer name property
+          featureInfo.properties = Object.assign(
+            { [LAYER_NAME_PROP]: f.layerName },
+            f.feature.props ?? {}
+          );
+          featureInfo.position = new Cartographic(longitude, latitude);
+
+          featureInfo.configureDescriptionFromProperties(f.feature.props);
+          featureInfo.configureNameFromProperties(f.feature.props);
+
+          featureInfos.push(featureInfo);
+        });
 
-            return featureInfo;
-          })
-      );
       // No view is set and we have geoJSON object
       // So we pick features manually
     } else if (
       this.source instanceof GeojsonSource &&
       this.source.geojsonObject
     ) {
+      // Get rough meters per pixel (at equator) for given zoom level
+      const zoomMeters = 156543 / Math.pow(2, level);
       // Create circle with 10 pixel radius to pick features
       const buffer = circle(
         [CesiumMath.toDegrees(longitude), CesiumMath.toDegrees(latitude)],
-        10 * this.terria.mainViewer.scale,
+        10 * zoomMeters,
```

Comment:
This change was required to fix share links with `pickedFeatures`.

When Terria was picking features on startup, `mainViewer.scale` was always equal to `1` - which meant sometimes features weren't being picked correctly.

Using the rough meters per pixel according to Web Mercator at the equator is not ideal - but the whole point of this is to allow some rough threshold to picking features without any mass (eg points or lines) - so I reckon it is ok

Reply:
If you don't trust my math - I sure don't - then you can verify using this table - https://wiki.openstreetmap.org/wiki/Zoom_levels